src/stravalib/client.py

```diff
@@ -638,12 +638,87 @@ def get_friend_activities(self, limit: int | None = None) -> NoReturn:
             "See https://developers.strava.com/docs/january-2018-update/"
         )
 
+    def _validate_activity_type(
+        self,
+        params: dict[str, Any],
+        activity_type: str | None,
+        sport_type: str | None,
+        create: bool = False,
+    ) -> dict[str, Any]:
+        """Validate activity type and sport type.
+
+        Parameters
+        ----------
+        params : dict
+            A dictionary of activity values used to update or create an
+            activity.
+        activity_type : str, default=None
+            The activity type (case-insensitive).
+            Deprecated. Prefer to use sport_type. In a request where both type
+            and sport_type are present, this field will be ignored.
+            See https://developers.strava.com/docs/reference/#api-models-UpdatableActivity.
+            For possible values see: :class:`stravalib.model.Activity.TYPES`
+        sport_type : str, default=None
+            For possible values (case-sensitive) see:
+            :class:`stravalib.model.Activity.SPORT_TYPES`
+        create : bool, default=False
+            Flag indicating if the activity is being created (or updated).
+
+        Returns
+        -------
+        dict
+            Dictionary containing overlapping parameter values between the two
+            methods
+        Raises
+        ------
+        ValueError
+            If the activity_type or sport_type is invalid.
+        """
+
+        if create and not sport_type:
+            raise ValueError(
+                "The sport_type key is required to create a Strava activity"
+            )
```

Comment:
Referring to my remark above, should we really raise an error here (as the Strava API, per its documentation, supports the `type` parameter)?